diff --git a/cat.py b/cat.py
index f879933..e74fa22 100644
--- a/cat.py
+++ b/cat.py
@@ -4,8 +4,11 @@ This program prints stdin to the screen.
 import sys
 
 def cat(file):
-    data = file.read()
-    sys.stdout.buffer.write(data)
+    while True:
+        chunk = file.read(8192)  # Read in 8KB chunks
+        if not chunk:
+            break
+        sys.stdout.buffer.write(chunk)
 
 if __name__ == "__main__":
     if len(sys.argv) > 1:
